components/api/core/api_impl.bal

```diff
@@ -18,24 +18,46 @@
 import cellery_hub_api/db;
 import ballerina/http;
 
-public function createOrgs (http:Request _createOrgsReq, gen:OrgCreateRequest _createOrgsBody) returns http:Response {
-    // stub code - fill as necessary
-    http:Response _createOrgsRes = new;
-    string _createOrgsPayload = "Sample createOrgs Response";
-    _createOrgsRes.setTextPayload(_createOrgsPayload);
-	return _createOrgsRes;
+public function createOrg (http:Request createOrgReq, gen:OrgCreateRequest createOrgsBody) returns http:Response{
+    http:Response _addOrgRes = new;
+    if (createOrgReq.hasHeader(USERNAME)){
+        string userName = createOrgReq.getHeader(USERNAME);
+        var res = db:insertOrganization(userName, createOrgsBody);
+        if (res is error) {
+            string errMsg = "Unexpected error occured while inserting organization " + untaint createOrgsBody.orgName;
+            log:printError(errMsg, err = res);
+            return buildErrorResponse(http:INTERNAL_SERVER_ERROR_500, API_ERROR_CODE, errMsg, "Organization name \'" + 
+                                                                    untaint createOrgsBody.orgName+ "\' is already taken");
+        } else {        
+            _addOrgRes.statusCode = http:OK_200;
+            log:printDebug("Organization \'" +createOrgsBody.orgName+ "\' is created. Author : " +userName);   
+            return _addOrgRes;
+        }
+    } else {
+        string errDescription = "Unauthenticated request : Username is not found";
```

Comment:
We can have a util function for this else task. This might be useful in all other places as well.